sql/src/main/java/io/crate/metadata/IngestionRuleInfos.java

```diff
@@ -0,0 +1,39 @@
+package io.crate.metadata;
+
+
+import io.crate.metadata.rule.ingest.IngestRulesMetaData;
+import org.elasticsearch.cluster.metadata.MetaData;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static java.util.Collections.emptyIterator;
+
+public class IngestionRuleInfos implements Iterable<IngestionRuleInfo> {
+
+    private final MetaData metaData;
+
+    public IngestionRuleInfos(MetaData metaData) {
+        this.metaData = metaData;
+    }
+
+    @Override
+    public Iterator<IngestionRuleInfo> iterator() {
+        IngestRulesMetaData ingestRulesMetaData = metaData.custom(IngestRulesMetaData.TYPE);
+        if (ingestRulesMetaData == null || ingestRulesMetaData.getIngestRules() == null) {
+            return emptyIterator();
+        }
+        List<IngestionRuleInfo> ingestionRuleInfoList = new ArrayList<>();
```

Comment:
I think this is a bit of unnecessary overhead to build a new list on *each* iterator call. This could be done once in the constructor - or we could return a view onto the map here without creating a new list.